File: sudoq-app/sudoqmodel/src/test/java/de/sudoq/model/sudoku/sudokuTypes/SymbolIteratorTest.java
package de.sudoq.model.sudoku.sudokuTypes;

import org.junit.Test;

import de.sudoq.model.sudoku.Position;
import de.sudoq.model.sudoku.complexity.Complexity;
import de.sudoq.model.sudoku.complexity.ComplexityConstraint;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertTrue;

public class SymbolIteratorTest {

	SudokuType sst;

	@Test
	public void test4x4(){
		sst = TypeBuilder.getType(SudokuTypes.standard4x4);
		testEquality(sst);
	}

	@Test
	public void test9x9(){
		sst = TypeBuilder.getType(SudokuTypes.standard9x9);
		testEquality(sst);
	}

	@Test
	public void test6x6(){
		sst = TypeBuilder.getType(SudokuTypes.standard6x6);
		testEquality(sst);
	}

	@Test
	public void test16x16(){
		sst = TypeBuilder.getType(SudokuTypes.standard16x16);
		testEquality(sst);
	}

	@Test
	public void testArbitrary(){
		sst = TypeBuilder.getType(SudokuTypes.standard4x4);
		for(int i : new int[]{1,2,3,20,100}){
			sst.setNumberOfSymbols(i);
			testEquality(sst);
		}
	}

	private void testEquality(SudokuType sst) {
		int counter = 0;
		for (int i:
			 sst.getSymbolIterator()) {
			assertEquals(counter++, i);
		}
		assertEquals(sst.getNumberOfSymbols(), counter);
		Position p = sst.getSize();
	}



}

File: sudoq-app/sudoqmodel/src/test/kotlin/de/sudoq/model/sudoku/sudokuTypes/SymbolIteratorTest.kt
package de.sudoq.model.sudoku.sudokuTypes

import de.sudoq.model.sudoku.Position
import org.amshove.kluent.`should be`
import org.amshove.kluent.`should not be null`
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class SymbolIteratorTest {

    @Test
    fun test4x4() {
        val sst = TypeBuilder.getType(SudokuTypes.standard4x4)
        testEquality(sst)
    }

    @Test
    fun test9x9() {
        val sst = TypeBuilder.getType(SudokuTypes.standard9x9)
        testEquality(sst)
    }

    @Test
    fun test6x6() {
        val sst = TypeBuilder.getType(SudokuTypes.standard6x6)
        testEquality(sst)
    }

    @Test
    fun test16x16() {
        val sst = TypeBuilder.getType(SudokuTypes.standard16x16)
        testEquality(sst)
    }

    @ParameterizedTest
    @ValueSource(ints = [1, 2, 3, 20, 100])
    fun arbitrary(numOfSymbols: Int) {
        val arbitrarySudokuType = SudokuType(SudokuTypes.standard4x4, numOfSymbols, 0f, Position[1,1],
            Position[1,1], ArrayList(), ArrayList(), ArrayList(), ComplexityConstraintBuilder())
        testEquality(arbitrarySudokuType)
    }

    private fun testEquality(sst: SudokuType?) {
        var counter = 0
        for (i in sst!!.symbolIterator) {
            i.`should be`(counter++)
        }
        sst.numberOfSymbols.`should be`(counter)
        sst.size.`should not be null`()
    }
}